shell.nix

```diff
@@ -0,0 +1,12 @@
+{ pkgs ? import <nixpkgs> {} }:
```

Comment:
A number of other NervesHub and Nerves projects use asdf and `.tool-versions`. I'll let the maintainers weigh in but I think following that standard might be preferable to adding nix as a one-off.